lib/collection/property.js

```diff
@@ -178,12 +189,13 @@ _.extend(Property, /** @lends Property */ {
         // extract all arguments to get the list of variables
         !_.isArray(variables) && (variables = _.tail(arguments));
 
-        return str.replace(this.REGEX_EXTRACT_VARS, function (match, token) {
-            var r = _.findValue(variables, token);
-            r && _.isFunction(r) && (r = r());
-            r && _.isFunction(r.toString) && (r = r.toString());
-            return nativeTypes[(typeof r)] ? r : match;
-        });
+        while (replaced && ++loopCounter < 21) {
```

Comment:
also, this `21` value is an important thing in this feature's context and as such difficult to configure and find. it should move out as a static variable in Property.